sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -232,114 +232,150 @@ CosmosAsyncContainer setId(String id) {
-     * failure the {@link Flux} will error.
+     * failure the {@link CosmosPagedFlux} will error.
      *
      * @param query   the query.
      * @param options the feed options.
-     * @return an {@link Flux} containing one or several feed response pages of the
+     * @return an {@link CosmosPagedFlux} containing one or several feed response pages of the
      *         obtained items or an error.
      */
-    public Flux<FeedResponse<CosmosItemProperties>> queryItems(String query, FeedOptions options) {
+    public CosmosPagedFlux<CosmosItemProperties> queryItems(String query, FeedOptions options) {
         return queryItems(new SqlQuerySpec(query), options);
     }
 
     /**
      * Query for documents in a items in a container
      *
-     * After subscription the operation will be performed. The {@link Flux} will
+     * After subscription the operation will be performed. The {@link CosmosPagedFlux} will
      * contain one or several feed response of the obtained items. In case of
-     * failure the {@link Flux} will error.
+     * failure the {@link CosmosPagedFlux} will error.
      *
      * @param querySpec the SQL query specification.
-     * @return an {@link Flux} containing one or several feed response pages of the
+     * @return an {@link CosmosPagedFlux} containing one or several feed response pages of the
      *         obtained items or an error.
      */
-    public Flux<FeedResponse<CosmosItemProperties>> queryItems(SqlQuerySpec querySpec) {
+    public CosmosPagedFlux<CosmosItemProperties> queryItems(SqlQuerySpec querySpec) {
         return queryItems(querySpec, null);
     }
 
     /**
      * Query for documents in a items in a container
      *
-     * After subscription the operation will be performed. The {@link Flux} will
+     * After subscription the operation will be performed. The {@link CosmosPagedFlux} will
      * contain one or several feed response of the obtained items. In case of
-     * failure the {@link Flux} will error.
+     * failure the {@link CosmosPagedFlux} will error.
      *
      * @param querySpec the SQL query specification.
      * @param options   the feed options.
-     * @return an {@link Flux} containing one or several feed response pages of the
+     * @return an {@link CosmosPagedFlux} containing one or several feed response pages of the
      *         obtained items or an error.
      */
-    public Flux<FeedResponse<CosmosItemProperties>> queryItems(SqlQuerySpec querySpec, FeedOptions options) {
+    public CosmosPagedFlux<CosmosItemProperties> queryItems(SqlQuerySpec querySpec, FeedOptions options) {
+        if (options == null) {
+            options = new FeedOptions();
+        }
+        FeedOptions feedOptions = options;
+        Flux<FeedResponse<CosmosItemProperties>> feedResponseFlux = queryItemsInternal(querySpec, feedOptions);
+        return new CosmosPagedFlux<>(feedResponseFlux::next, (continuationToken) -> {
```

Comment:
This will result in huge perf issue on our side.

a constructor need to be added to CosmosPagedFlux and hence PageFluxBase which accept a Flux<FeedResponse>